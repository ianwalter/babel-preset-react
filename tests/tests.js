const { test } = require('@ianwalter/bff')
const { transformSync } = require('@babel/core')
const preset = require('..')

test('basic config', ({ expect }) => {
  const src = `
    import React, { lazy } from 'react'
    import ReactDOM from 'react-dom'

    const msg = lazy(() => import('./Msg'))

    ReactDOM.render(
      <h1>{msg}</h1>,
      document.getElementById('app')
    )
  `
  const { code } = transformSync(src, { presets: [preset] })
  expect(code).toMatchSnapshot()
})
