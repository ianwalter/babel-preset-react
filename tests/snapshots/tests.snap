// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic config 1`] = `
"import \\"core-js/modules/es.object.to-string\\";
import \\"core-js/modules/es.promise\\";
import React, { lazy } from 'react';
import ReactDOM from 'react-dom';
var msg = lazy(function () {
  return import('./Msg');
});
ReactDOM.render(React.createElement(\\"h1\\", null, msg), document.getElementById('app'));"
`;
